### PR TITLE
BAU: Initialisers refactor (2nd attempt)

### PR DESCRIPTION
**Pull request by vixus0**
- Consolidate small but related initialisers.
- Keep the default Rails skeleton initialisers for easier tracking
  through Rails versions.
- Some typo and code-style changes.

Author: @vixus0

### CODE REVIEW

File: config/initializers/constants.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FINGERPRINT_CONFIG = '/assets2/fp.gif'.freeze
+SESSION_VALIDATOR = SessionValidator.new(Integer(CONFIG.session_cookie_duration))
+SAML_PROXY_HOST = CONFIG.saml_proxy_host
+METADATA_CLIENT = MetadataClient.new(SAML_PROXY_HOST)
```
Comment (chrisholmes) [inline config/initializers/constants.rb]: I think the old arrangement for these constants is more appropriate as these things don't belong together.

**Comment by vixus0** [comment]
This gets rid of some of the constants but creating things in ApplicationController might not be what we want to do.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -98,6 +98,10 @@ def session_validator
     SESSION_VALIDATOR
   end
 
+  def metadata_client
+    METADATA_CLIENT
+  end
+
```
Comment (chrisholmes) [inline app/controllers/application_controller.rb]: I think this can be moved to the MetadataController

**Comment by chrisholmes** [comment]
Happy to merge after my last comment is addressed!

**Comment by vixus0** [comment]
Done!

**Pull request merged**